include/clap/factory/draft/preset-discovery.h

```diff
@@ -0,0 +1,321 @@
+   // otherwise they must be null.
+   //
+   // The subpath defines a human friendly path to the preset in the container file.
+   // It must be unique within the container file.
+   //
+   // The load_key is a machine friendly string used to load the preset inside the container via a
+   // the preset-load plug-in extension. The load_key can also just be the subpath if that's what
+   // the plugin wants but it could also be some other unique id like a database primary key or a
+   // binary offset. It's use is entirely up to the plug-in.
+   //
+   // If the function returns false, the the provider must stop calling back into the receiver.
+   bool(CLAP_ABI *begin_preset)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                const char                                           *name,
+                                const char                                           *subpath,
+                                const char                                           *load_key);
+
+   // Adds a plug-in id that this preset can be used with.
+   // plugin_abi: 0 for CLAP
+   void(CLAP_ABI *add_plugin_id)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                 uint32_t                                              plugin_abi,
+                                 const char                                           *plugin_id);
+
+   // Sets the sound pack to which the preset belongs to.
+   void(CLAP_ABI *set_soundpack_id)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                    const char *soundpack_id);
+
+   // Sets the flags, see clap_preset_discovery_flags.
+   // If unset, they are then inherited from the location.
+   void(CLAP_ABI *set_flags)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                             uint32_t                                              flags);
+
+   // Adds a creator name for the preset.
+   void(CLAP_ABI *add_creator)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                               const char                                           *creator);
+
+   // Sets a description of the preset.
+   void(CLAP_ABI *set_description)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                   const char *description);
+
+   // Sets the creation time and last modification time of the preset.
+   // The timestamps are in seconds since UNIX EPOCH, see C's time_t time(time_t *).
+   // If one of the time isn't known, then set it to 0.
+   // If this function is not called, then the indexer may look at the file's creation and
+   // modification time.
+   void(CLAP_ABI *set_timestamps)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                  uint64_t creation_time,
+                                  uint64_t modification_time);
```

Comment:
Maybe we could introduce a new 'clap_timestamp' type for these UNIX EPOCH timestamps? Then the UNIX EPOCH explanation could be put next to this type and wouldn't have to be mentioned everywhere.
Or maybe call the type 'clap_unix_epoch_timestamp' so that the correct usage is clarified immediately.

Reply:
I‘d like that, maybe with helper functions

Reply:
Yes good idea!